perfkitbenchmarker/spark_service.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Benchmarking support for Apache Spark services.
+
+In order to benchmark Apache Spark services such as Google Cloud
+Platform's Dataproc or Amazon's EMR, we create a BaseSparkService
+class.  Classes to wrap each provider's Apache Spark Service are
+in the provider directory as a subclass of BaseSparkService.
+
+Also in this module is a PkbSparkService, which builds a Spark
+cluster by creating VMs and installing the necessary software.
+
+For more on Apache Spark: http://spark.apache.org/
+"""
+
+import abc
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import resource
+
+FLAGS = flags.FLAGS
+
+# Cloud to use for pkb-created Spark service.
+PKB_MANAGED = 'pkb_managed'
+PROVIDER_MANAGED = 'managed'
+
+_SPARK_SERVICE_REGISTRY = {}
+
+
+def GetSparkService(cloud):
+  """Get the Spark class corresponding to 'cloud'."""
+  if cloud in _SPARK_SERVICE_REGISTRY:
+    return _SPARK_SERVICE_REGISTRY.get(cloud)
+  else:
+    registered_clouds = ' '.join(_SPARK_SERVICE_REGISTRY.keys())
+    raise Exception('No Spark service found for {0} Options are {1}'.format(
```

Comment:
Given that only a subset of clouds that support spark services may be loaded, I'm not sure you should use the service registry as a list of all options